src/libs/antares/study/variable-print-info.cpp

```diff
@@ -76,161 +79,141 @@ variablePrintInfoCollector::variablePrintInfoCollector(AllVariablesPrintInfo* al
 }
 
 void variablePrintInfoCollector::add(const AnyString& name,
-                                     uint nbGlobalResults,
                                      uint dataLevel,
                                      uint fileLevel)
 {
-    allvarsinfo->add(new VariablePrintInfo(name, nbGlobalResults, dataLevel, fileLevel));
+    allvarsinfo->add(name.to<std::string>(), VariablePrintInfo(dataLevel, fileLevel));
 }
 
 // ============================================================
 // All variables print information
 // ============================================================
-AllVariablesPrintInfo::AllVariablesPrintInfo() :
- maxColumnsCount(0), numberSelectedAreaVariables(0), numberSelectedLinkVariables(0)
-{
-    // Re-initializing the iterator
-    it_info = allVarsPrintInfo.begin();
-}
 
-AllVariablesPrintInfo::~AllVariablesPrintInfo()
+static std::string to_uppercase(std::string& str)
 {
-    clear();
+    std::transform(str.begin(), str.end(), str.begin(), ::toupper);
+    return str;
 }
 
-void AllVariablesPrintInfo::add(VariablePrintInfo* v)
+void AllVariablesPrintInfo::add(std::string name, VariablePrintInfo v)
 {
-    allVarsPrintInfo.push_back(v);
+    std::string upperCaseName = to_uppercase(name);
+    if (not exists(upperCaseName))
+    {
+        index_to_name[(unsigned int)allVarsPrintInfo.size()] = upperCaseName;
+        allVarsPrintInfo.insert(std::pair<std::string, VariablePrintInfo>(upperCaseName, v));
```

Comment:
Use `try_emplace`, see https://www.fluentcpp.com/2018/12/11/overview-of-std-map-insertion-emplacement-methods-in-cpp17/

Reply:
Done